packages/front/src/chess/chess.lib.ts

```diff
@@ -0,0 +1,114 @@
+import { BLACK, CHESS_BOARD_SIZE, WHITE } from './chess.constants'
+import {
+  EChessPiece,
+  TChessBoard,
+  TChessCastlingAbility,
+  TChessPieceColor,
+  TChessSquare,
+  TChessSquareIndex,
+  TChessState
+} from './chess.models'
+
+export function fenToChessPiece(fenPiece: EChessPiece): EChessPiece {
+  return fenPiece
+}
+
+export function fenToCastlingAbility(
+  fenCastling: string
+): TChessCastlingAbility {
+  return {
+    WHITE: {
+      kingSide: fenCastling.includes(EChessPiece.WhiteKing),
+      queenSide: fenCastling.includes(EChessPiece.WhiteQueen)
+    },
+    BLACK: {
+      kingSide: fenCastling.includes(EChessPiece.BlackKing),
+      queenSide: fenCastling.includes(EChessPiece.BlackQueen)
+    }
+  }
+}
+
+export function fenToEnPassantTargetSquareIndex(
+  fen: string
+): TChessSquareIndex | null {
+  if (fen === '-') return null
+  const [file, rank, ...rest] = fen.split('')
+  const parsedFile = parseFile(file)
+  const parsedRank = parseRank(rank)
+  if (rest.length > 1 || !parsedFile || !parsedRank) return null
+  return squareToIndex({ file: parsedFile, rank: parsedRank })
+}
```

Comment:
Can you explain what you are doing here?

Reply:
Yeaaah, maybe there is some refactoring ahead. I have failed the laws of clean code :D!

It converts a fen [en passant target square](https://www.chessprogramming.org/Forsyth-Edwards_Notation#En_passant_target_square) into a index that can be used to select the given square from the chess board which is represented as a 1d array in the frontend.

And if there is no target square to parse, (the string is "-"), then there is no square to target.



The reason the board is represented as a basic 1d array instead of a 2d matrix is just performance. And the reason why we return an index instead of, lets say, a object containing x and y coordinates is also just performance.

Modifying a 2d grid is quite a bit more slow than just using a single array. I read through some other chess repositories for research and this was something some of them did.

It might need an abstraction layer over it to make it more readable. The current datastructures don't necessarily scream readability as of now :p